packages/juggler/src/types/array.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Type, AnyType} from './type';
+
+/**
+ * Array type, such as string[]
+ */
+export class ArrayType<T> implements Type<Array<T>> {
```

Comment:
AFAICT, this class is not covered by unit-tests at all.

I am also wondering, how is this new `Type<T>` system integrated with existing juggler code? Is it a good thing that methods like `isInstance` and `isCoercible` are implemented here in loopback-next and not in the juggler?